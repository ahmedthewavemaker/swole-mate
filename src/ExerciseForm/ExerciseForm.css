*{
    box-sizing: border-box;
}

.tabs{
   padding: 5px 3px;
    border-style: solid;
}

h3{
    padding: 12px;
}

li{
    list-style: none;
}

.nav{
    list-style: none;
    display: inline-flex;
}

.exerciseList{
    font-size: large;
    font-weight: bold;
}