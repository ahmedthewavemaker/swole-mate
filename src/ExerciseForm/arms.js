import React from 'react'


function arms() {
    return (
        <div>
            <ol >
                <li> Arms workout 1------Sets: 3------Reps: 10</li>
                <li> Arms workout 2------Sets: 3------Reps: 10</li>
                <li> Arms workout 3------Sets: 3------Reps: 10</li>
                <li> Arms workout 4------Sets: 3------Reps: 10</li>
                <li> Arms workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default arms;