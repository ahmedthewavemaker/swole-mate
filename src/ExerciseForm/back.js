import React from 'react'


function back() {
    return (
        <div>
            <ol >
                <li> Back workout 1------Sets: 3------Reps: 10</li>
                <li> Back workout 2------Sets: 3------Reps: 10</li>
                <li> Back workout 3------Sets: 3------Reps: 10</li>
                <li> Back workout 4------Sets: 3------Reps: 10</li>
                <li> Back workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default back;