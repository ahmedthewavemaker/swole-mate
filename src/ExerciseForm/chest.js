import React from 'react'


function chest() {
    return (
        <div>
            <ol>
                <li> Chest workout 1------Sets: 3------Reps: 10</li>
                <li> Chest workout 2------Sets: 3------Reps: 10</li>
                <li> Chest workout 3------Sets: 3------Reps: 10</li>
                <li> Chest workout 4------Sets: 3------Reps: 10</li>
                <li> Chest workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default chest;