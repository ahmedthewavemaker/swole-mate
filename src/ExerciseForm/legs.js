import React from 'react'


function legs() {
    return (
        <div>
            <ol>
                <li> Legs workout 1------Sets: 3------Reps: 10</li>
                <li> Legs workout 2------Sets: 3------Reps: 10</li>
                <li> Legs workout 3------Sets: 3------Reps: 10</li>
                <li> Legs workout 4------Sets: 3------Reps: 10</li>
                <li> Legs workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default legs;