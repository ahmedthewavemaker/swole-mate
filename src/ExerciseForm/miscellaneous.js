import React from 'react'


function miscellaneous() {
    return (
        <div>
            <ol>
                <li> Miscellaneous workout 1------Sets: 3------Reps: 10</li>
                <li> Miscellaneous workout 2------Sets: 3------Reps: 10</li>
                <li> Miscellaneous workout 3------Sets: 3------Reps: 10</li>
                <li> Miscellaneous workout 4------Sets: 3------Reps: 10</li>
                <li> Miscellaneous workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default miscellaneous;