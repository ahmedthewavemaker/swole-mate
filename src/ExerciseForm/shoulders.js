import React from 'react'


function shoulders() {
    return (
        <div>
            <ol >
                <li> Shoulders workout 1------Sets: 3------Reps: 10</li>
                <li> Shoulders workout 2------Sets: 3------Reps: 10</li>
                <li> Shoulders workout 3------Sets: 3------Reps: 10</li>
                <li> Shoulders workout 4------Sets: 3------Reps: 10</li>
                <li> Shoulders workout 5------Sets: 3------Reps: 10</li>
            </ol>
        </div>
)

}

export default shoulders;