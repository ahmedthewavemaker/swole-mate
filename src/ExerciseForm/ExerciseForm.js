import React, { Component } from 'react';
import './ExerciseForm.css'
import { NavLink } from 'react-router-dom';
import AppContext from '../AppContext';
import config from '../config'



class ExerciseForm extends Component {

    state = {
        exercises: [],

    }




    static contextType = AppContext



    componentDidMount() {
        this.getWorkouts();

    }

    //3 functions of the requests (GET, POST, and DELETE)
    getWorkouts = () => {
        const email = this.context.email


        fetch(`${config.API_ENDPOINT}/api/workout/?email=${email}`)
            .then(res => res.json())
            .then(exercises => {

                this.setState({ exercises });
            })


    }

    postWorkout = (event) => {
        event.preventDefault();
        const { exercise, sets, reps } = event.target;

        const newWorkout = {
            email: this.context.email,
            muscle: this.props.match.params.muscle,
            exercise: exercise.value,
            reps: reps.value,
            sets: sets.value
        }

        fetch(`${config.API_ENDPOINT}/api/workout`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json'
            },
            body: JSON.stringify(newWorkout)
        })
            .then(response => {
                this.getWorkouts()

            })
            .catch(error => {
                console.error(error)
            })
    }

    deleteWorkouts = (id) => {

        fetch(`${config.API_ENDPOINT}/api/workout/${id}`, {
            method: 'DELETE',

        })
           
            .then(exercises => {

                this.getWorkouts()
            })


    }


    render() {

        return (
            <div>
                <div>
                    <header>
                        <h1> Swole Mate </h1>
                    </header>


                    <div >
                        <ul className="nav">
                            <li><h3> <NavLink to="/exercise/chest">Chest</NavLink></h3></li>
                            <li><h3> <NavLink to="/exercise/back"> Back</NavLink></h3></li>
                            <li><h3> <NavLink to="/exercise/shoulders">Shoulders</NavLink></h3></li>
                            <li><h3> <NavLink to="/exercise/arms">Arms</NavLink></h3></li>
                            <li><h3> <NavLink to="/exercise/legs">Legs</NavLink></h3></li>
                            <li><h3> <NavLink to="/exercise/miscellaneous">Miscellaneous</NavLink></h3></li>
                        </ul>
                        <div>
                            <ol className='exerciseList'>
                                <li> {this.props.match.params.muscle} </li>
                            </ol>
                            <form onSubmit={this.postWorkout}>
                                <label htmlFor='exercise'>New Workout: </label>
                                <input name='exercise' id='exercise' type='text' />

                                <label htmlFor='sets'>Sets:  </label>
                                <input name='sets' id='sets' type='number' />

                                <label htmlFor='reps'>Reps:  </label>
                                <input name='reps' id='reps' type='number' />
                                <button className='buttonz'>Add Workout</button>

                            </form>

                            {this.state.exercises.filter(e => e.muscle == this.props.match.params.muscle).map(exercise => {
                                console.log(exercise)
                                return (


                                    <li className='exerciseList'>

                                    Workout: {exercise.exercise}----
                                    Sets: {exercise.sets}----  
                                     Reps: {exercise.reps}----
                                     Muscle: {exercise.muscle}----

                                        <button className='buttonz' onClick={e => this.deleteWorkouts(exercise.id)}>Delete Workout</button>
                                    </li>
                                )
                            })}

                        </div>

                    </div>
                </div>
            </div>
        )
    }
}
ExerciseForm.defaultProps={
    match: {
        params: {}
    }
}

export default ExerciseForm;
