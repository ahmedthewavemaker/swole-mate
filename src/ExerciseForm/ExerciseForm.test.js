import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ExerciseForm from './ExerciseForm';
import { MemoryRouter } from 'react-router-dom';

describe('ExerciseForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter> <ExerciseForm />
      </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter> <ExerciseForm />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


});
