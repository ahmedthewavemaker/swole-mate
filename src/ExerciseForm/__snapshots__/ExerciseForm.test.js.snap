// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseForm component renders the UI as expected 1`] = `
Array [
  " ",
  <div>
    <div>
      <header>
        <h1>
           Swole Mate 
        </h1>
      </header>
      <div>
        <ul
          className="nav"
        >
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/chest"
                onClick={[Function]}
              >
                Chest
              </a>
            </h3>
          </li>
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/back"
                onClick={[Function]}
              >
                 Back
              </a>
            </h3>
          </li>
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/shoulders"
                onClick={[Function]}
              >
                Shoulders
              </a>
            </h3>
          </li>
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/arms"
                onClick={[Function]}
              >
                Arms
              </a>
            </h3>
          </li>
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/legs"
                onClick={[Function]}
              >
                Legs
              </a>
            </h3>
          </li>
          <li>
            <h3>
               
              <a
                aria-current={null}
                href="/exercise/miscellaneous"
                onClick={[Function]}
              >
                Miscellaneous
              </a>
            </h3>
          </li>
        </ul>
        <div>
          <ol
            className="exerciseList"
          >
            <li>
               
               workout------ Sets: 3------Reps: 10
            </li>
          </ol>
          <form
            onSubmit={[Function]}
          >
            <label
              htmlFor="exercise"
            >
              New Workout: 
            </label>
            <input
              id="exercise"
              name="exercise"
              type="text"
            />
            <label
              htmlFor="sets"
            >
              Sets: 
            </label>
            <input
              id="sets"
              name="sets"
              type="number"
            />
            <label
              htmlFor="reps"
            >
              Reps: 
            </label>
            <input
              id="reps"
              name="reps"
              type="number"
            />
            <button>
              Add Workout
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;
